File: server/internal/dto/category.go
package dto

// ============ 通用分类请求 ============

// ListCategoryRequest 分类列表请求
type ListCategoryRequest struct {
	Page     int `form:"page" binding:"omitempty,min=1"`
	PageSize int `form:"page_size" binding:"omitempty,min=1,max=1000"`
}

// ============ 通用分类响应 ============

// CategoryForWebResponse 分类响应
type CategoryForWebResponse struct {
	ID          uint   `json:"id"`
	Name        string `json:"name"`
	Slug        string `json:"slug"`
	URL         string `json:"url"`
	Description string `json:"description"`
	Count       int    `json:"count"`
	Sort        int    `json:"sort"`
}

// ============ 后台分类管理请求 ============

// CreateCategoryRequest 创建分类请求
type CreateCategoryRequest struct {
	Name        string `json:"name" binding:"required,max=50"`
	Description string `json:"description" binding:"max=200"`
	Sort        int    `json:"sort"`
}

// UpdateCategoryRequest 更新分类请求
type UpdateCategoryRequest struct {
	Name        string `json:"name" binding:"required,max=50"`
	Description string `json:"description" binding:"max=200"`
	Sort        int    `json:"sort"`
}

File: admin/src/types/category.ts
/*
项目名称：JeriBlog
文件名称：category.ts
创建时间：2026-04-16 15:08:10

系统用户：Jerion
作　　者：Jerion
联系邮箱：416685476@qq.com
功能描述：类型定义 - category类型
*/

// 分类实体
export interface Category {
  id: number
  name: string
  slug: string
  description: string
  count: number
  sort: number
}

// 分页数据
export interface CategoryListData {
  list: Category[]
  total: number
  page: number
  page_size: number
}